src/esm_runscripts/filedicts.py

```diff
@@ -314,10 +317,39 @@ def _resolve_paths(self) -> None:
         for key, path in self.locations.items():
             self[f"absolute_path_in_{key}"] = path.joinpath(self[f"name_in_{key}"])
 
-    def _path_type(self, path: pathlib.Path) -> Union[str, bool]:
+    def convert_to_path(self, path: Union[str, pathlib.Path]) -> pathlib.Path:
+        """
+        Converts the ``path`` to pathlib.Path object since ESM-Tools uses pathlib instead of str. This is a low level function to be called before higher level file I/O functions
+
+        Parameters
+        ----------
+        path : str, pathlib.Path
+            path to be checked
+
+        Returns
+        -------
+        path : pathlib.Path
+            converted path
+
+        Raises
+        ------
+        TypeError
+            - When input is not string or pathlike object. Eg. dict, integer, ...
+        """
+        if isinstance(path, pathlib.Path):
+            return path
+        if isinstance(path, str):
+            return pathlib.Path(path)
+        if not isinstance(path, (str, pathlib.Path)):
+            datatype = type(path).__name__
+            raise TypeError(
+                f"Path ``{path}`` has an incompatible datatype ``{datatype}``. str or pathlib.Path is expected"
+            )
```

Comment:
If you want to put the actual input the user had given into the exception:
```suggestion
        try:
            return pathlib.Path(path)
        except TypeError:
            raise EsmToolsFileDictError(f"Unable to convert {path} from {type(path)} to a pathlib.Path")
```

Reply:
Removed it